qoi.h

```diff
@@ -657,7 +657,7 @@ void *qoi_read(const char *filename, qoi_desc *desc, int channels) {
 		return NULL;
 	}
 
-	bytes_read = fread(data, 1, size, f);
+	bytes_read = (int)fread(data, 1, size, f);
```

Comment:
`fread` returns `ssize_t`. Casting to `int` truncates to 2 GiB. Change `size_t size, bytes_read;` above instead.

Also this call to `fread` is lacking any error checking.